packages/dds/tree/src/test/simple-tree/arrayNode.spec.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import {
+	ImplicitAllowedTypes,
+	InsertableTreeFieldFromImplicitField,
+	InsertableTreeNodeFromImplicitAllowedTypes,
+	NodeKind,
+	SchemaFactory,
+	TreeArrayNode,
+	TreeFieldFromImplicitField,
+	TreeNodeSchema,
+} from "../../simple-tree/index.js";
+
+import { hydrate } from "./utils.js";
+import { Mutable } from "../../util/index.js";
+
+type ArrayNodeSchema<T extends ImplicitAllowedTypes> = TreeNodeSchema<
+	string,
+	NodeKind.Array,
+	TreeArrayNode<T>,
+	Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>>,
+	true
+>;
+
+const schemaFactory = new SchemaFactory("Test");
+
+function hydrateArray<T extends TreeNodeSchema | readonly TreeNodeSchema[]>(
```

Comment:
I think this should be named "TContent" since its not super obvious that this doesn't take an array schema like if you were using hydrate directly. Also is there a reason not to use ImplicitAllowedTypes here?

Reply:
TContent sounds good. ImplicitAllowedTypes doesn't compile. I couldn't really figure out why, so I went with this which I think will allow anything other than recursive types. If you want to help me debug how to get ImplicitAllowedTypes working, message me.